### PR TITLE
qemu: Rename ring buffer to com.canonical.lxd

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

File: lxd-agent/main_agent.go [hunk 1]
```diff
@@ -262,8 +262,8 @@ func (c *cmdAgent) startStatusNotifier(ctx context.Context, chConnected <-chan s
 
 // writeStatus writes a status code to the vserial ring buffer used to detect agent status on host.
 func (c *cmdAgent) writeStatus(status string) error {
-	if shared.PathExists("/dev/virtio-ports/org.linuxcontainers.lxd") {
-		vSerial, err := os.OpenFile("/dev/virtio-ports/org.linuxcontainers.lxd", os.O_RDWR, 0600)
+	if shared.PathExists("/dev/virtio-ports/com.canonical.lxd") {
+		vSerial, err := os.OpenFile("/dev/virtio-ports/com.canonical.lxd", os.O_RDWR, 0600)
```
Comment (simondeziel) [inline lxd-agent/main_agent.go:266]: That will require a corresponding fix in https://github.com/lxc/distrobuilder/blob/main/distrobuilder/main.go#L634

**Review by simondeziel — COMMENTED** [review]
@monstermunchkin are there any implication to applying this change for long running VMs?

@escabo should `com.canonical.lxd` be used or something else like `com.ubuntu.lxd`?

**Comment by monstermunchkin** [comment]
> @monstermunchkin are there any implication to applying this change for long running VMs?

I just tested this on an Alpine VM, and it's causing problems with lxd-agent. The image has `/dev/virtio-ports/org.linuxcontainers.lxd` hard-coded in the lxd-agent setup script.

The official Ubuntu images have the same problem. There, the path is hard-coded in `/lib/systemd/system/lxd-agent.service`.

Perhaps we should leave this for now.

**Comment by simondeziel** [comment]
Yeah, I feel this one will be tricky to rename. A possible way would be to ship a `lxd-agent` update that renames the existing symlink to `com.canonical.lxd` during post-install and have the systemd unit updated with `ConditionPathExists=/dev/virtio-ports/com.canonical.lxd`.

However that would need to be backported to all supported releases and be coordinated carefully. Even then, I'm not sure it would work.

**Comment by tomponline** [comment]
Something for lxd 6.0 perhaps

**Comment by tomponline** [comment]
@monstermunchkin please could you create an issue for this?

**Pull request closed without merging**